src/containers/coursePageDetails/index.jsx

```diff
@@ -0,0 +1,139 @@
+      numberOfEnrolledStudents,
+      authors,
+      language,
+      profits,
+      courseReviews,
+    } = this.state.course;
+    const { isLoading, showReviewsNum, error } = this.state;
+
+    return (
+      <main className={styles.coursePageDetailsWrapper}>
+        {isLoading ? (
+          <Spinner />
+        ) : error ? (
+          <div className={styles.error}>{error.message}</div>
+        ) : (
+          <div>
+            <h2 className={styles.courseTitle}>{courseName}</h2>
+            <p className={styles.shortDescription}>{description}</p>
+            <div className={styles.courseStats}>
+              <div className={styles.stars}>
+                <StarRatings
+                  rating={rating}
+                  starRatedColor="#ff8c05"
+                  starEmptyColor="#515151"
+                  starDimension="24px"
+                  starSpacing="3px"
+                  numberOfStars={5}
+                  name="courseRating"
+                />
+              </div>
+              <span>{rating}</span>
+              <span className={styles.verticalLine}></span>
+              <span>
+                <font>Students:</font> {numberOfEnrolledStudents}
+              </span>
+            </div>
+            <div>
+              <span>
+                <font>Author: </font>
+                {authors}
+              </span>
+              <span className={styles.verticalLine}></span>
+              <span>
+                <font>Language: </font>
+                {language}
+              </span>
+            </div>
+            <hr />
+            <p className={styles.profitTitle}>Your profit</p>
+            <ul className={styles.profitList}>
+              {profits.map(el => (
+                <li key={el.id}>{el.description}</li>
+              ))}
+            </ul>
+            <hr />
+            <p className={styles.reviewsTitle}>Reviews</p>
+            {courseReviews.slice(0, showReviewsNum).map((el, key) => (
+              <div className={styles.reviewWrapper} key={key}>
+                <div>
+                  <span>Jhon Doe(Default)</span>
+                  <div className={styles.stars}>
+                    <StarRatings
+                      rating={el.rating}
+                      starRatedColor="#ff8c05"
+                      starEmptyColor="#515151"
+                      starDimension="24px"
+                      starSpacing="3px"
+                      numberOfStars={5}
+                      name={`review${el.key}`}
+                    />
+                  </div>
+                  <span>{el.createdAt}</span>
+                </div>
+                <p>{el.text}</p>
+              </div>
+            ))}
+            {courseReviews.length > showReviewsNum && (
+              <div className={styles.buttonWrapper}>
+                <button
+                  className={styles.showMoreButton}
+                  type="button"
+                  onClick={() => this.showMoreReviews()}>
```

Comment:
What?? Please, You need to wrap this.showMoreReviews into the external function. It's need to optimize the performance of code